content/300-guides/200-deployment/201-serverless/400-deploy-to-aws-lambda.mdx

```diff
@@ -37,7 +47,65 @@ In the `generator` block of your `schema.prisma` file, add the following:
+
+In the case of AWS lambda, you will want to have the following line in your `.env` file
+
+```env file=.env
+PRISMA_CLI_BINARY_TARGETS=native,rhel-openssl-1.0.x
+```
+
+<Admonition>
+
+`prisma migrate` is a command in the `prisma` package. Normally, this package is installed as a dev dependency. Depending on your setup, you may need to install this package as a dependency instead in order for the command to be available in AWS Lambda
+
+</Admonition>
+
+### Connection pooling
+
+Generally, when you use a FaaS (Function as a Service) environment to interact with a database, every function invocation can result in a new connection to the database. This is not a problem with a constantly running Node.js server. Therefore, it is beneficial to pool DB connections to get better performance. You can use [Accelerate](/data-platform/accelerate) to solve this issue. For other solutions, see the [connection management guide for serverless environments](/guides/performance-and-optimization/connection-management#serverless-environments-faas).
+
+## AWS SAM guidance
+
+This section covers resolving issues when using the AWS SAM framework. Check the other sections for notes on [AWS Lambda in general](#general-aws-lambda-guidance), [Serverless Framework](#serverless-framework-guidance), or [SST](#sst-guidance).
+
+### Loading environment variables
+
+AWS SAM does not support loading values from a `.env` file directly. You will need to use "AWS Native" ways of storing and retrieving these parameters. [This guide](https://medium.com/bip-xtech/a-practical-guide-to-surviving-aws-sam-d8ab141b3d25) provides a great overview of your options and how to store and retrieve values in Parameters, SSM, Secrets Manager, and more.
+
+### Loading required files
+
+AWS SAM supports [esbuild](https://esbuild.github.io/) in order to bundle your TypeScript code. However, the full esbuild API is not exposed and esbuild plugins are not supported. This leads to problems when using Prisma in your application as certain files (like `schema.prisma`) must be available at runtime.
+
+To get around this, we will directly reference needed files in our code so that they are correctly bundled. In your application, you could add the following lines to your application where Prisma is instantiated.
+
+```ts file=app.ts
+import schema from './prisma/schema.prisma'
+import x from './node_modules/.prisma/client/libquery_engine-rhel-openssl-1.0.x.so.node'
+
+if (process.env.NODE_ENV !== 'production') {
+  console.debug(schema, x)
+}
+```
+
+You will also need to tell esbuild how to bundle these files by adding the following lines to `Metadata.BuildProperties` in your `template.yaml`:
+
+```yaml file=template.yaml
+Loader:
+  - .prisma=file
+  - .so.node=file
+AssetNames: '[name]'
```

Comment:
What does `[name]` do here? Is that to be replaced with a name of your choice?

Reply:
If you don't had `AssetNames` the name of the file is given a quasi-random string to the end. So something like `schema.prisma` becomes `schema-JQ7XRO21.prisma` so this is needed to make sure that the files can be found.

Reply:
Is `name` a placeholder here? If so, how do people know which value to replace it with?

Reply:
It's a placeholder value, yes, but its understood by `esbuild` https://esbuild.github.io/api/#asset-names

It's very confusing and AWS SAM does not have great documentation on this. [It _should_ be documented here](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-using-build-typescript.html#serverless-sam-cli-using-build-typescript-metadata) but it's not.